app/code/community/Lesti/Fpc/Helper/Data.php

```diff
@@ -258,4 +259,30 @@ public function getContentType(\Mage_Core_Controller_Response_Http $response)
 
         return 'text/html; charset=UTF-8';
     }
+
+    /**
+     * Transform array parameter to string to avoid error and make correct validation in regexp pattern
+     *
+     * @param mixed $param Input parameter
+     *
+     * @return mixed
+     */
+    protected function _paramToString($param)
+    {
+        if (is_array($param)) {
+            $param = array_reduce(
+                $param,
+                function ($carry, $item) {
```

Comment:
I guess this can lead to conflicts if we just concat all items of the array. What about just using [serialize](http://php.net/manual/en/function.serialize.php) instead?